db/records/operations/insert.py

```diff
@@ -76,10 +76,15 @@ def insert_records_from_csv(table, engine, csv_filepath, column_names, header, d
                     cursor.copy_expert(copy_sql, temp_file)
 
 
-def insert_from_select(from_table, target_table, engine, mappings=None):
-    target_table_col_list = [col for col in target_table.c if not MathesarColumn.from_column(col).is_default]
+def insert_from_select(from_table, target_table, engine, sa_col_mappings=None):
+    if sa_col_mappings:
+        # unzipping column mappings into list of respective tables
+        from_table_col_list, target_table_col_list = zip(*sa_col_mappings)
+    else:
+        from_table_col_list = [col for col in from_table.c if not MathesarColumn.from_column(col).is_default]
+        target_table_col_list = [col for col in target_table.c if not MathesarColumn.from_column(col).is_default]
     with engine.begin() as conn:
-        sel = select([col for col in from_table.c if not MathesarColumn.from_column(col).is_default])
+        sel = select(from_table_col_list)
```

Comment:
@mathemancer during the execution of this part of the code the select statement generates a warning when the `from_table_col_list` from the if block is passed, 
SAWarning: SELECT statement has a cartesian product between FROM element(s) "public.books2" and FROM element "public.books2".  Apply join condition(s) between each element to resolve.
This then causes an error in the formation of the SQL statement to be executed:
```sql
ERROR:  table name "books2" specified more than once
STATEMENT:  INSERT INTO public.books (title, author) SELECT public.books2.name, public.books2.author_name FROM public.books2, public.books2
```
Notice the extra `public.books2` at the end of the SQL statement.

The `from_table_col_list` from the else block works as expected.

Reply:
Each column from the `Column` model carries its own SQLAlchemy `Table` instance. This is a problem, since SQLAlchemy can't determine automatically that those table objects refer to the same underlying database table. Thus, it generates a `FROM` statement for each.

You should use the names from the columns and get them from the passed table instance instead (as you do in the `else` block).

Reply:
That makes sense, I was wondering why that behavior was happening, as in printing the column instance also showed no change `(Column('name', TEXT(), table=<books2>), Column('author_name', TEXT(), table=<books2>))`.